kotlin-analysis/src/main/kotlin/org/jetbrains/dokka/analysis/EnvironmentAndFacade.kt

```diff
@@ -51,7 +51,10 @@ class DokkaMessageCollector(private val logger: DokkaLogger) : MessageCollector
 }
 
 // It is not data class due to ill-defined equals
-class EnvironmentAndFacade(val environment: KotlinCoreEnvironment, val facade: DokkaResolutionFacade) {
+class EnvironmentAndFacade(val environment: KotlinCoreEnvironment, val facade: DokkaResolutionFacade, private val analysisEnvironment: AnalysisEnvironment) {
     operator fun component1() = environment
     operator fun component2() = facade
+    protected fun finalize() {
+        analysisEnvironment.dispose()
+    }
```

Comment:
oof, there's a great deal of problems and uncertainties with `finalize`, but the biggest one is that it's deprecated for removal, so it will fail sooner or later: https://openjdk.org/jeps/421

Is it possible to close it explicitly? Make the class implement `Closeable` and wrap usages in `.use {}` or close it some other way when the generation has been completed? 

This could even be an extension point or a feature where plugins register closeable components, and they are all closed at the end of documentation generation or as soon as documentable model has been built (probably somewhere in `SingleModuleGeneration#generate`)